test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/lang/CharacterTest.scala

```diff
@@ -429,6 +429,50 @@ class CharacterTest {
     assertEquals(65339, Character.toLowerCase(65339))
   }
 
+/*
+def format(codePoint: Int): String = "0x%04x".format(codePoint)
+
+for(cp <- 0 until Character.MAX_CODE_POINT) {
+  val titleCaseCP: Int = Character.toTitleCase(cp)
+  val upperCaseCP: Int = Character.toUpperCase(cp)
+
+  if (titleCaseCP != upperCaseCP) {
+    if (cp <= Character.MAX_VALUE) {
+      assert(titleCaseCP <= Character.MAX_VALUE)
+      println(s"    assertEquals('${titleCaseCP.toChar}', Character.toTitleCase('${cp.toChar}'))")
+    }
+
+    println(s"    assertEquals(${format(titleCaseCP)}, Character.toTitleCase(${format(cp)}))")
+  }
+}
+*/
+  @Test def toTitleCase(): Unit  = {
```

Comment:
We should definitely also test cases where the result is the same as `toUpperCase`. Basically you can add all the tests written for `toUpperCase` (including the ones that get "aborted" because the result does not fit in 1 char/1 code point) but for `toTitleCase`.

Reply:
FYI, this logic got updated based upon additional testing (well the `toUpperCase(Char)` got fixed) but yes I still should add tests here,  I just didn't want you to overlook the implementation change to `def toTitleCase(ch: scala.Char): scala.Char`